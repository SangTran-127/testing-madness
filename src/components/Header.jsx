import React from 'react'

const Header = ({ title }) => {
    return (
        <nav>
            <h1>{title}</h1>
            <ul>
                <li>home</li>
                <li>about</li>
                <li>register</li>
            </ul>
        </nav>
    )
}

export default Header