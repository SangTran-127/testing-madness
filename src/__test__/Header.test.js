import renderer from "react-test-renderer";
import Header from "../components/Header";
import { render, screen } from "@testing-library/react";
test("header should render title", () => {
  render(<Header title="my blog" />);
  const headerNav = screen.getByText(/my blog/i);
  expect(headerNav).toBeInTheDocument();
});
test("header should render li", () => {
  render(<Header title="my blog" />);
  const li = screen.getAllByRole("listitem");
  expect(li.length).toBe(3);
});
test("test snapshot header", () => {
  const snapshotHeader = renderer.create(<Header title="Sang" />).toJSON();
  expect(snapshotHeader).toMatchSnapshot();
});
