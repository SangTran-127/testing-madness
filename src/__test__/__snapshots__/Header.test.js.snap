// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot header 1`] = `
<nav>
  <h1>
    Sang
  </h1>
  <ul>
    <li>
      home
    </li>
    <li>
      about
    </li>
    <li>
      register
    </li>
  </ul>
</nav>
`;
